### PR TITLE
Reimplemented the plugin for bless-4.0.0

### PR DESCRIPTION
**Pull request by TiddoLangerak**
I also adopted the code styling from the main project and moved everything to the BlessCSS repo. 

@BlessCSS/owners - Could you take a look at this and provide feedback where needed?

### CODE REVIEW

**Comment by mtscout6** [comment]
This looks fantastic! The only missing link here are unit tests, your demo is serving as a manual test, though automated unit/integration tests would really help for maintainability.

**Comment by aabenoja** [comment]
We do have an active issue to reimplement the imports in the parser, and I like what you're doing here to generate them. Maybe not right now, but I would like to see this logic moved over and we can just use a bless option to trigger imports.

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		source : sourceFileName
+	};
+	const blessed = chunk(source, options);
+
+	const chunks = blessed.data.map((chunk, index) => {
+		let name;
+		if (index === blessed.data.length - 1) {
+			name = fileName;
+			const imports = generateImports(fileName, blessed.data.length - 1) ;
+			chunk = `${imports}\n${chunk}`;
```
Comment (mtscout6) [inline src/index.js:40]: Doesn't this affect sourcemaps?

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+	return {
+		apply(compiler) {
+			compiler.plugin('this-compilation', compilation => {
+				compilation.plugin('optimize-assets', (assets, callback) => {
+
+					Object.keys(assets)
+						.filter(assetName => /\.css$/.test(assetName))
+						.map(fileName => [fileName, assets[fileName].source()])
+						.map(([fileName, source]) => createChunks(fileName, source, sourceMaps))
```
Comment (mtscout6) [inline src/index.js:70]: These map calls can be reduced to:

```
.map(fileName => createChunks(fileName, assets[fileName].source(), sourceMaps))
```

File: src/index.js (hunk 1 above)
Comment (TiddoLangerak) [inline src/index.js:40]: Yes, you're right, completely missed that. I'll get it fixed.

**Comment by TiddoLangerak** [comment]
I agree that it might be better to move the logic over to the main project. I didn't realize that imports where completely missing still from the main project, they were present in 3.x and I just assumed that only this API method didn't generate them. I'll try to see if I can create a PR for this for the main project this weekend.

**Comment by aabenoja** [comment]
It was on our todo list, we just never had the time when working on the `css-band-aid` fork.

**Comment by karlhorky** [comment]
Would be cool if we could add the functionality from #4 as well.

I can take a shot at this if you would like some help.

**Comment by thomaswmanion** [comment]
Is this getting merged ever?

**Comment by SergTanchenko** [comment]
@mtscout6 do you plan to update bless version from 3.0.3 to 4.0.4? :)

**Comment by mtscout6** [comment]
There are other members of the org that can do this. I don't personally have the time right now. @BlessCSS/owners @BlessCSS/contributors

**Pull request closed without merging**